### ISSUE TITLE
Blackscreen after #308

### ISSUE THREAD

**Issue by jld3103** [go-flutter-desktop/go-flutter#326]
@pchampio
After #308 my app only shows a blackscreen and spits out this error on linux (also tested windows with wine and got the same error):
```
2019/12/26 12:59:30 error.go:171: PlatformError: GLX: Failed to make context current
```
When running using `hover run -b '@2d582358f34c257933c47e1f0591fc38c713c78c'` (the previous commit) the app starts like normal so it's definitely an issue with #308.

**Comment by pchampio** [comment]
What's your flutter version?
Does this error happen with:
```
❯ flutter --version
Flutter 1.12.13+hotfix.5 • channel stable • https://github.com/flutter/flutter
Framework • revision 27321ebbad (2 weeks ago) • 2019-12-10 18:15:01 -0800
Engine • revision 2994f7e1e6
Tools • Dart 2.7.0
```

**Comment by jld3103** [comment]
I'm currently on:
```
Flutter 1.13.6-pre.25 • channel master • https://github.com/flutter/flutter
Framework • revision b7af5e714d (7 hours ago) • 2019-12-25 23:49:55 -0800
Engine • revision 33813929e3
Tools • Dart 2.8.0 (build 2.8.0-dev.0.0 886615d0f9)
```
and it also happens on stable

**Comment by pchampio** [comment]
@jld3103 does `hover run -b '@no_resize_crash_v2'` works?

**Comment by jld3103** [comment]
Nope same error

**Comment by pchampio** [comment]
Just updated to glfw v3.3, does this fixes your issue ?

**Comment by jld3103** [comment]
No, didn't work

**Comment by pchampio** [comment]
What about: `hover run -b '@no_resize_crash_v2' --opengl=none` ??

**Comment by jld3103** [comment]
Still the same

**Comment by pchampio** [comment]
I'm clueless

**Comment by pchampio** [comment]
Can you share the full output of the `hover run`?

**Comment by pchampio** [comment]
FWIW, this should work: https://github.com/flutter/engine/blob/e731ad9b709df5d65f44e0703a5362beae186b18/shell/platform/embedder/tests/embedder_unittests.cc#L2585-L2661

**Comment by jld3103** [comment]
```
$ hover run -b '@no_resize_crash_v2'
hover: Using engine from cache
hover: Cleaning the build directory
hover: Bundling flutter app
hover: Downloading 'go-flutter' @no_resize_crash_v2
go: finding github.com/go-flutter-desktop/go-flutter no_resize_crash_v2
go: finding github.com/go-gl/glfw/v3.3/glfw latest
go: finding github.com/go-gl/gl latest
hover: 'go-flutter' is on version: v0.34.1-0.20191227143434-8b348a820a54
hover: Compiling 'go-flutter' and plugins
hover: Successfully compiled
hover: Build finished, starting app...
hover: Running ginko in debug mode
flutter: Observatory listening on http://127.0.0.1:50300/lVMud-OzKxQ=/
go-flutter: calculated pixelRatio limited to a minimum of 1.0. metrics: {"dpi":90.21564482029598,"monitorWidthMm":473,"monitorWidthSc":1680,"mscpi":90.21564482029598,"pixelRatio":0.5638477801268499,"ppsc":1,"windowWidthPx":800,"windowWidthSc":800}
go-flutter: no handler found for channel flutter/isolate
2019/12/27 15:53:38 error.go:171: PlatformError: GLX: Failed to make context current
2019/12/27 15:53:38 plugin.go:219: dbpath= /home/jld3103/.config/de.ginko/ginko/libCachedImageData.db
Syncing files to device Flutter test device...                         ⢿2019/12/27 15:53:39 error.go:171: PlatformError: GLX: Failed to make context current                          ⣯2019/12/27 15:53:39 error.go:171: PlatformError: GLX: Failed to make context current
2019/12/27 15:53:39 error.go:171: PlatformError: GLX: Failed to make context current                                                                                                  ⣷2019/12/27 15:53:39 error.go:171: PlatformError: GLX: Failed to make context current
2019/12/27 15:53:39 error.go:171: PlatformError: GLX: Failed to make context current                                                                                                  ⣾2019/12/27 15:53:39 error.go:171: PlatformError: GLX: Failed to make context current                                                                                                  ⣻2019/12/27 15:53:39 error.go:171: PlatformError: GLX: Failed to make context current
2019/12/27 15:53:39 error.go:171: PlatformError: GLX: Failed to make context current                                                                                                  ⡿2019/12/27 15:53:39 error.go:171: PlatformError: GLX: Failed to make context current                                                                                                  ⣟2019/12/27 15:53:39 error.go:171: PlatformError: GLX: Failed to make context current
2019/12/27 15:53:39 error.go:171: PlatformError: GLX: Failed to make context current
2019/12/27 15:53:40 error.go:171: PlatformError: GLX: Failed to make context current
2019/12/27 15:53:40 error.go:171: PlatformError: GLX: Failed to make context current                                                                                                  ⣯2019/12/27 15:53:40 error.go:171: PlatformError: GLX: Failed to make context current
2019/12/27 15:53:40 error.go:171: PlatformError: GLX: Failed to make context current
2019/12/27 15:53:40 error.go:171: PlatformError: GLX: Failed to make context current                                                                                                  ⣷2019/12/27 15:53:40 error.go:171: PlatformError: GLX: Failed to make context current                                                                                                  ⣾2019/12/27 15:53:40 error.go:171: PlatformError: GLX: Failed to make context current                                                                                                  ⣽2019/12/27 15:53:40 error.go:171: PlatformError: GLX: Failed to make context current
2019/12/27 15:53:40 error.go:171: PlatformError: GLX: Failed to make context current
2019/12/27 15:53:40 error.go:171: PlatformError: GLX: Failed to make context current
2019/12/27 15:53:40 error.go:171: PlatformError: GLX: Failed to make context current                                                                                                  ⣻2019/12/27 15:53:40 error.go:171: PlatformError: GLX: Failed to make context current
2019/12/27 15:53:40 error.go:171: PlatformError: GLX: Failed to make context current
2019/12/27 15:53:40 error.go:171: PlatformError: GLX: Failed to make context current                                                                                                  ⢿2019/12/27 15:53:40 error.go:171: PlatformError: GLX: Failed to make context current                                                                                                  ⡿2019/12/27 15:53:40 error.go:171: PlatformError: GLX: Failed to make context current
2019/12/27 15:53:40 error.go:171: PlatformError: GLX: Failed to make context current                                                                                                   
(This is taking an unexpectedly long time.)       ⣟2019/12/27 15:53:40 error.go:171: PlatformError: GLX: Failed to make context current
2019/12/27 15:53:40 error.go:171: PlatformError: GLX: Failed to make context current
2019/12/27 15:53:40 error.go:171: PlatformError: GLX: Failed to make context current
2019/12/27 15:53:40 error.go:171: PlatformError: GLX: Failed to make context current                                                (This is taking an unexpectedly long time.)       ⣾2019/12/27 15:53:41 error.go:171: PlatformError: GLX: Failed to make context current
2019/12/27 15:53:41 error.go:171: PlatformError: GLX: Failed to make context current
2019/12/27 15:53:41 error.go:171: PlatformError: GLX: Failed to make context current                                                23,506ms (!)                                       

🔥  To hot reload changes while running, press "r". To hot restart (and rebuild state), press "R".
An Observatory debugger and profiler on Flutter test device is available at: http://127.0.0.1:50300/lVMud-OzKxQ=/
For a more detailed help message, press "h". To detach, press "d"; to quit, press "q".
```

**Comment by jld3103** [comment]
I tried running using `@latest`, but also got the error. So somehow your revert doesn't work?

**Comment by pchampio** [comment]
`@latest` uses the latest tag available, try `@master`

**Comment by pchampio** [comment]
might be related to: https://github.com/go-flutter-desktop/go-flutter/issues/68

**Comment by pchampio** [comment]
Do you have a multi GPU setup?

**Comment by jld3103** [comment]
No

**Comment by davidmartos96** [comment]
I was having the same problem as @jld3103 and using the master branch worked. Flutter v1.12.13

**Comment by pchampio** [comment]
@davidmartos96 on which go-flutter version are you running?

**Comment by davidmartos96** [comment]
@pchampio 
v0.34.0 => Fails
v0.34.1-0.20191227132350-c4c03d13f0e6 => Works

**Comment by XANOZOID** [comment]
Also having the same problem, on Ubuntu bionic. I have nvidia graphics drivers installed if that makes a difference. 

I'll have to use the work-around suggested `hover run -b '@2d582358f34c257933c47e1f0591fc38c713c78c'` until this is fixed. . . 

I'm using the flutter beta-channel, since flutter-master seems to be breaking web and this project allows me to work on the two on the same flutter branch. 

I tried running `hover run -b '@no_resize_crash_v2'` and can also report there is still a black screen with the same PlatformError.

**Comment by pchampio** [comment]
@davidmartos96 on v0.34.1-0.20191227132350-c4c03d13f0e6 the PR was reverted.

I'll keep the bad build `v0.34.0` as the `@latest` for a small amount of time in order to see how much people are affected by this.

If it happen that you have this issue, please try:
`hover run -b '@2d582358f34c257933c47e1f0591fc38c713c78c'` -> should work,
`hover run -b '@v0.34.0'` -> shouldn't work,
`hover run -b '@no_resize_crash_v2'` -> shouldn't work.
And post a comment on the issue thread.
Thanks for your contribution.

I'll report an issue on the flutter issue tracker once I have more time, I don't think it related to go-flutter.

**Comment by moffatman** [comment]
I am experiencing this same issue on Windows and Intel integrated graphics. The same workaround works for me.

**Comment by huage2580** [comment]
hover run -b '@2d582358f34c257933c47e1f0591fc38c713c78c'
work for me

**Comment by Murtuzakabul** [comment]
It is still not working for me. 
hover run -b '@2d582358f34c257933c47e1f0591fc38c713c78c' isn't working. I am getting the same black screen on windows.

Should I revert back to previous version ?

**Comment by pchampio** [comment]
> Should I revert back to previous version ?

Yes! [`v0.33.0`](https://github.com/go-flutter-desktop/go-flutter/releases/tag/v0.33.0)

**Comment by JoeyGaojingxing** [comment]
I met the black screen error just now, and I change to version `v0.33.0` which works well.
but I met another error which made me unable to use hot-reload

my flutter version 
```
Flutter 1.12.13+hotfix.6 • channel beta • https://github.com/flutter/flutter.git
Framework • revision 18cd7a3601 (4 weeks ago) • 2019-12-11 06:35:39 -0800
Engine • revision 2994f7e1e6
Tools • Dart 2.7.0
```

the error msg
```
go-flutter: no handler found for channel flutter/isolate
flutter: Observatory server failed to start after 2 tries
flutter: Observatory server failed to start after 3 tries
flutter: Observatory server failed to start after 4 tries
flutter: Observatory server failed to start after 5 tries
flutter: Observatory server failed to start after 6 tries
flutter: Observatory server failed to start after 7 tries
flutter: Observatory server failed to start after 8 tries
flutter: Observatory server failed to start after 9 tries
flutter: Observatory server failed to start after 10 tries
flutter: Observatory server failed to start after 11 tries
flutter: Could not start Observatory HTTP server:
SocketException: Failed to create server socket (OS Error: 以一种访问权限不允许的方式做了一个访问套接字的尝试。
, errno = 10013), address = 127.0.0.1, port = 50300
#0      _NativeSocket.bind (dart:io-patch/socket_patch.dart:644:7)
<asynchronous suspension>
#1      _RawServerSocket.bind (dart:io-patch/socket_patch.dart:1262:26)
#2      _ServerSocket.bind (dart:io-patch/socket_patch.dart:1522:29)
#3      ServerSocket._bind (dart:io-patch/socket_patch.dart:1513:26)
#4      ServerSocket.bind (dart:io/socket.dart:336:27)
#5      _HttpServer.bind (dart:_http/http_impl.dart:2692:25)
#6      HttpServer.bind (dart:_http:229:19)
#7      Server.startup.poll (dart:vmservice_io/server.dart:401:36)
<asynchronous suspension>
#8      Server.startup (dart:vmservice_io/server.dart:413:23)
<asynchronous suspension>
#9      main (dart:vmservice_io:262:12)
```

**Comment by pchampio** [comment]
@JoeyGaojingxing this issue is for the black-screen.

nevertheless, please try: `hover run --observatory-port 44333`

**Comment by JoeyGaojingxing** [comment]
Sorry about that, I just wanna report the blackscreen bug

**Comment by GeertJohan** [comment]
I also ran into this, and someone in the gitter chat seems to have the same issue as well.
I've just tagged the current master to v0.34.1 to revert this regression (which was introduced by #308) for all users.

**Comment by pchampio** [comment]
I need you guys.

The one that have the blackscreen error, can you share your OS platform? (window/linux/darwin)
Can you also try the new version: `hover run -b '@no_resize_crash_v2'`?
(I'm only targeting this fix at the darwin users, as the inital issue was only affecting them)

**Comment by jld3103** [comment]
On linux the fix works now.

**Comment by GeertJohan** [comment]
Works on my linux machine as well

**Comment by pchampio** [comment]
I want to do more investigation on this: https://github.com/flutter/flutter/issues/48651

**Comment by pchampio** [comment]
Closing this issue as the black-screen doesn't appear anymore. Please follow #133 for more information about the Segmentation fault: 11 when resizing window on MacOS

**Comment by ekasetiawans** [comment]
Sorry i'm late. I still facing this issue on my windows machine. But application showing black screen in release mode (hover build).

**Comment by pchampio** [comment]
@ekasetiawans #308 was reverted, our issue doesn't come from here.
Humm, after seeing your screen record, it's weird, I'm not sure the issue comes from go-flutter...
Anyway, can you open another issue, with more information.
I would like to have the output of `hover doctor` AND could you also run your app using the windows cmd, instead of double clicking the app to open it, please launch your app from the terminal and share the terminal output with us.
Also please provide us with a small sample that reproduce your issue, thanks.

**Comment by ekasetiawans** [comment]
I'm currently using flutter master channel. I trying to using flutter stable channel. I will share the result.

**Comment by ekasetiawans** [comment]
Solved by switching to flutter stable channel.

**Comment by pchampio** [comment]
@jld3103 
do you get the same issue with: `hover run -b '@macOS_no_resize_crash'` #390

**Comment by jld3103** [comment]
Works for me. No blackscreen or anything at all
